aws/hll/aws-s3-transfer-manager/src/io/error.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+use std::error::Error as StdError;
+use std::fmt;
+use std::fmt::Formatter;
+use std::io::{Error as StdIoError, ErrorKind as StdIoErrorKind};
+use tokio::task::JoinError;
+
+#[derive(Debug)]
+pub(crate) enum ErrorKind {
+    UpperBoundSizeHintRequired,
+    OffsetGreaterThanFileSize,
+    TaskFailed(JoinError),
+    IOError(StdIoError),
```

Comment:
even tho it look like `lo` I think we should use this casing per our convention:
```suggestion
    IoError(StdIoError),
```